main/src/main/java/org/mobilitydata/gtfsvalidator/validator/FeedServiceDateValidator.java

```diff
@@ -35,9 +38,19 @@
   @Override
   public void validate(NoticeContainer noticeContainer) {
     for (GtfsFeedInfo feedInfo : feedInfoTable.getEntities()) {
-      if (feedInfo.hasFeedStartDate()
-          && feedInfo.hasFeedEndDate()
-          && feedInfo.feedStartDate().isAfter(feedInfo.feedEndDate())) {
+      if (!feedInfo.hasFeedStartDate() || !feedInfo.hasFeedEndDate()) {
+        if (feedInfo.hasFeedStartDate() && !feedInfo.hasFeedEndDate()) {
+          noticeContainer.addValidationNotice(
+              new MissingRequiredFieldError(
+                  feedInfoTable.gtfsFilename(), feedInfo.csvRowNumber(), "feed_end_date"));
+        } else if (!feedInfo.hasFeedStartDate() && feedInfo.hasFeedEndDate()) {
+          noticeContainer.addValidationNotice(
+              new MissingRequiredFieldError(
+                  feedInfoTable.gtfsFilename(), feedInfo.csvRowNumber(), "feed_start_date"));
+        }
+        return;
+      }
+      if (feedInfo.feedStartDate().isAfter(feedInfo.feedEndDate())) {
```

Comment:
Hmmm, maybe you added the above IF statement to avoid having the `hasXXX()` methods here? IMHO it's easier to read and makes more sense if you keep this IF statement as it was originally written and remove the redundant IF above for that case.